src/sql/src/func.rs

```diff
@@ -744,6 +744,21 @@ impl ParamList {
                         return None;
                     }
                 }
+                // As of today, with no UDFs, the only operators that accept records/row types as
```

Comment:
Ah, this makes me pretty uncomfortable, though I hear you that without more complex impls, not super worthwhile. However, it's relatively few LOC to determine base equality of record types, so let me see if I can type up the implementation.